sns_ik_lib/test/sns_ik_math_utils_test.cpp

```diff
@@ -410,6 +442,61 @@ TEST(sns_ik_math_utils, pseudoInverse_damped_test)
 
 /*************************************************************************************************/
 
+TEST(sns_ik_math_utils, solveLinearSystem_fullRank_test)
+{
+  // test parameters
+  double tol = 1e-10;  // tolerance for matrix equality check
+  int nTest = 25;
+  int s1 = 84658;  // seed for the integer RNG
+  int s2 = 63541;  // seed for the floating point RNG
+  double low = -2.0;  double upp = 2.0;  // bounds on values in the A matrix
+  // test setup
+  Eigen::MatrixXd A, b;  // test linear system
+  Eigen::MatrixXd x;  // result
+  int rank; // rank of A, computed by linear solver
+  double err;  // residual error in the solution
+  for (int iTest = 0; iTest < nTest; iTest++) {
+    int n = sns_ik::rng_util::getRngInt(s1, 3, 9);
+    int m = sns_ik::rng_util::getRngInt(0, n, n + 5);
```

Comment:
n <= m is always the case here, which means we are looking for a least square solution. The use case for pseudo inverse in IK is to find a least norm solution where n > m. I just wonder both cases are tested here.

Reply:
I updated the code to check all shaped matrices (n<m, n=m, n<m)